code/game/objects/items/melee/f13onehanded.dm

```diff
@@ -274,24 +274,52 @@ obj/item/melee/onehanded/knife/switchblade
 		hitsound = 'sound/weapons/genhit.ogg'
 		sharpness = SHARP_NONE
 
+//////////////////
+// Cosmic Knife //
+////////////////// now you do the cleaning & heating manually.
+
 /obj/item/melee/onehanded/knife/cosmicdirty
 	name = "dirty cosmic knife"
-	desc = "A high-quality kitchen knife made from Saturnite alloy."
+	desc = "A high-quality kitchen knife made from Saturnite alloy, this one is covered in oxidation. Perhaps Abraxo might clean it up?"
 	icon_state = "knife_cosmic_dirty"
 	item_state = "knife"
 	force = 15
 	throwforce = 10
 	armour_penetration = 0.2
 
+// Abraxo my beloved. Can now be used directly to clean the blade.
+/obj/item/melee/onehanded/knife/cosmicdirty/attackby(obj/item/C, mob/user, params)
+	if(istype(C, /obj/item/crafting/abraxo))
+		user.visible_message("[user] begins cleaning the blade of the [src].", \
+				"<span class='notice'>You begin scrubbing the [src] with Abraxo. Smells nice.</span>", \
+				"<span class='italics'>You hear faint sounds of scrubbing.</span>")
+		playsound(get_turf(src), 'sound/FermiChem/heatdam.ogg', 50, TRUE)
+		if(!do_after(user, 60, TRUE, src))
+			return
+		new /obj/item/melee/onehanded/knife/cosmic(drop_location())
+		qdel(src)
+		qdel(C)
+	return TRUE
```

Comment:
```suggestion
		qdel(C)
		return
	return ..()
```

Reply:
Same for all procs below.
Might also want to make it automatically put the new item in hand.